ql/src/java/org/apache/hadoop/hive/ql/udf/generic/AbstractGenericUDFArrayBase.java

```diff
@@ -34,6 +35,7 @@
 public abstract class AbstractGenericUDFArrayBase extends GenericUDF {
 
     static final int ARRAY_IDX = 0;
+    static final int ELEMENT_IDX = 1;
```

Comment:
I guess it is more flexible if `checkValueAndListElementTypes` accepts an index than we hardcode the index here.
For example, `array_insert` of Spark SQL receives an element in the 3rd argument, not the 2nd.
https://spark.apache.org/docs/latest/api/sql/index.html#array_insert

Another option is we keep only `checkValueAndListElementTypes(arrayElementOI, valueOI)`, removing `checkValueAndListElementTypes(arguments, index?)`.

Reply:
Yes, keeping only checkValueAndListElementTypes(arrayElementOI, valueOI, elementIndex) method